### PR TITLE
Create rule S6494: C++ formatting functions should be used instead of C printf-like functions

### PR DESCRIPTION
**Pull request by tomasz-kaminski-sonarsource**

### CODE REVIEW

File: rules/S6494/cfamily/rule.adoc [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+`std::formatter` specializations.
+
+This rule raises issues for calls of the `sprintf` and `snprintf` functions that can be replaced by C++ formatting functions.
+
+== Noncompliant Code Example
+
+[source,cpp]
+----
+void printFunc(FILE* f, char* out, size_t n) {
+  sprintf(out, "%u %s", 10u, “text”);
```
Comment (abbas-sabra-sonarsource) [inline rules/S6494/cfamily/rule.adoc:23]: add Noncompilant comments

File: rules/S6494/cfamily/rule.adoc [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Secondly, the relation between the type and format specifier is more abstract.
+In particular, `{:d}` can be used to format any integer type, regardless of its size and signedness.
+Similarly, `{:f}` works for any floating point type.
+Furthermore, `{}` can be used for any type with default format spec, which makes it usable in the generic context.
+
+Finally, the `std::format` API can be extended to support custom types with the dedicated format specification via
+`std::formatter` specializations.
+
+This rule raises issues for calls of the `sprintf` and `snprintf` functions that can be replaced by C++ formatting functions.
```
Comment (abbas-sabra-sonarsource) [inline rules/S6494/cfamily/rule.adoc:16]:
```suggestion
This rule raises issues for calls of the `sprintf` and `snprintf` functions that can be replaced by the C++ formatting functions.
```

File: rules/S6494/cfamily/rule.adoc [hunk 3]
```diff
@@ -0,0 +1,63 @@
+In contrast to C printf-like functions, the `std::format` family of formatting functions provides
+a safer, and more robust interface for performing text formatting.
+
+Firstly, C++ formatting facilities perform validation of the format string against the type
+of the formatted argument. If the validation fails, it is reported as compilation error
```
Comment (abbas-sabra-sonarsource) [inline rules/S6494/cfamily/rule.adoc:5]:
```suggestion
of the formatted argument. If the validation fails, it is reported as a compilation error
```

File: rules/S6494/cfamily/rule.adoc (hunk 1 above)
Comment (tomasz-kaminski-sonarsource) [reply rules/S6494/cfamily/rule.adoc]: Added

**Pull request merged**